geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/LRemDUnitTest.java

```diff
@@ -0,0 +1,184 @@
+    jedis.lpush(key, elementList.toArray(new String[] {}));
+
+    // Remove all elements except for ELEMENT_TO_CHECK
+    final int uniqueElementsCount = initialListSize / uniqueElements;
+    assertThat(jedis.lrem(key, 0, makeElementString(key, 0))).isEqualTo(uniqueElementsCount);
+    assertThat(jedis.lrem(key, -uniqueElementsCount, makeElementString(key, 1)))
+        .isEqualTo(uniqueElementsCount);
+
+    clusterStartUp.crashVM(1); // kill primary server
+
+    assertThat(jedis.llen(key)).isEqualTo(uniqueElementsCount);
+    assertThat(jedis.lrem(key, uniqueElementsCount, makeElementString(key, 2)))
+        .isEqualTo(uniqueElementsCount);
+    assertThat(jedis.exists(key)).isFalse();
+  }
+
+
+  @Test
+  public void givenBucketsMoveDuringLrem_thenOperationsAreNotLost() throws Exception {
+    AtomicBoolean running = new AtomicBoolean(true);
+
+    List<String> listHashtags = makeListHashtags();
+    String key1 = makeListKeyWithHashtag(1, listHashtags.get(0));
+    String key2 = makeListKeyWithHashtag(2, listHashtags.get(1));
+    String key3 = makeListKeyWithHashtag(3, listHashtags.get(2));
+
+    List<String> elementList1 = makeListWithRepeatingElements(key1);
+    List<String> elementList2 = makeListWithRepeatingElements(key2);
+    List<String> elementList3 = makeListWithRepeatingElements(key3);
+
+    jedis.lpush(key1, elementList1.toArray(new String[] {}));
+    jedis.lpush(key2, elementList2.toArray(new String[] {}));
+    jedis.lpush(key3, elementList3.toArray(new String[] {}));
+
+    Future<Integer> future1 =
+        executor.submit(() -> performLremAndVerify(key1, running, elementList1));
+    Future<Integer> future2 =
+        executor.submit(() -> performLremAndVerify(key2, running, elementList2));
+    Future<Integer> future3 =
+        executor.submit(() -> performLremAndVerify(key3, running, elementList3));
+
+    for (int i = 0; i < 50; i++) {
+      clusterStartUp.moveBucketForKey(listHashtags.get(i % listHashtags.size()));
+      Thread.sleep(500);
+    }
+
+    running.set(false);
+
+    verifyLremResult(key1, future1.get());
+    verifyLremResult(key2, future2.get());
+    verifyLremResult(key3, future3.get());
+  }
+
+  private void verifyLremResult(String key, int iterationCount) {
+    for (int i = UNIQUE_ELEMENTS - 1; i >= iterationCount; i--) {
+      String element = makeElementString(key, i);
+      assertThat(jedis.lrem(key, COUNT_OF_UNIQUE_ELEMENT, element))
+          .isEqualTo(COUNT_OF_UNIQUE_ELEMENT);
+    }
+    assertThat(jedis.exists(key)).isFalse();
+  }
+
+  private Integer performLremAndVerify(String key, AtomicBoolean isRunning, List<String> list) {
+    assertThat(jedis.llen(key)).isEqualTo(LIST_SIZE_FOR_BUCKET_TEST);
+    int count = COUNT_OF_UNIQUE_ELEMENT;
+
+    int iterationCount = 0;
+    while (isRunning.get()) {
+      count = -count;
+      String element = makeElementString(key, iterationCount);
+      assertThat(jedis.lrem(key, count, element)).isEqualTo(COUNT_OF_UNIQUE_ELEMENT);
```

Comment:
Could we have some more assertions here, confirming that the contents of the list is correct after each LREM? Maybe checking the size of the list is as expected, and that the list doesn't contain the elements that have just been removed, using `lrange()`? That way if something goes wrong, we can fail faster and return an error immediately afterwards, rather than relying on a final verification step once all the operations have finished.